packages/analytics-js-plugins/__tests__/deviceModeDestinations/index.test.ts

```diff
@@ -381,6 +381,287 @@ describe('DeviceModeDestinations Plugin', () => {
       // Should fallback to loadOnlyIntegrations
       expect(mockFilterDestinations).toHaveBeenCalledWith({ All: true }, expect.any(Array));
     });
+
+    // disabled destination filtering
+    describe('Disabled Destination Filtering', () => {
+      it('should only enabled destination be set to active destinations', () => {
+        const mixedDestinations: Destination[] = [
+          {
+            ...mockDestinations[0]!,
+            enabled: true, // Should be included
+          },
+          {
+            ...mockDestinations[1]!,
+            enabled: false, // Should be filtered out
+          },
+          {
+            id: 'dest4',
+            displayName: 'GA4',
+            userFriendlyId: 'GA4___dest4',
+            enabled: false, // Should be filtered out
+            shouldApplyDeviceModeTransformation: true,
+            propagateEventsUntransformedOnError: false,
+            config: {
+              apiKey: 'key4',
+              blacklistedEvents: [],
+              whitelistedEvents: [],
+              eventFilteringOption: 'disable' as const,
+            },
+          },
+        ];
+
+        mockState.nativeDestinations.configuredDestinations.value = mixedDestinations;
+
+        plugin.nativeDestinations.setActiveDestinations(
+          mockState,
+          mockPluginsManager,
+          mockErrorHandler,
+          mockLogger,
+        );
+
+        expect(state.nativeDestinations.activeDestinations.value?.every(dest => dest.enabled)).toBe(
+          true,
+        );
+      });
```

Comment:
This is not conclusive that dest[0] is included in this list.

We should ensure that activeDestinations exactly contains a single item and it is dest[0].

Reply:
Please update other tests also if needed.

Reply:
Addressed